db/migrate/20211126012045_add_user_to_posts.rb

```diff
@@ -0,0 +1,5 @@
+class AddUserToPosts < ActiveRecord::Migration[6.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
```

Comment:
Ok so weird rails thing, it doesn't play nice with `foreign_key`. So when creating a reference like this set `foreign_key: false`. I know that seems weird cause obviously it is being used as a foreign key but rails will still treat it as one anyway, it just won't be set as one in the database.